src/Decode.fs

```diff
@@ -402,54 +369,68 @@ module Decode =
             Ok None
         else
             match decoder path value with
-            | Ok v -> Ok(Some v)
-            | Error er -> Error er
+            | Ok v -> Ok <| Some v
+            | Error err -> Error err
 
-    let optional (fieldName : string) (decoder : Decoder<'value>) : Decoder<'value option> =
+    let optional (fieldName: string) (decoder: Decoder<'value>) : Decoder<'value option> =
         fun path value ->
             if Helpers.isObject value then
-                let fieldValue = Helpers.getField fieldName value
-                if Helpers.isUndefined fieldValue then Ok None
-                else decodeMaybeNull (path + "." + fieldName) decoder fieldValue
+                match Helpers.getField fieldName value with
+                | Some field ->
+                    if Helpers.isUndefined field then
```

Comment:
💡 I actually don't think I need this `if` block since `Helpers.getField` won't return `Undefined`, it will return `None`